sdk/src/cache/cache_client.rs

```diff
@@ -304,6 +306,52 @@ impl CacheClient {
         request.send(self).await
     }
 
+    /// Sets a batch of items in a Momento Cache
+    ///
+    /// # Arguments
+    ///
+    /// * `cache_name` - name of cache
+    /// * `items` - HashMap of items to set
+    ///
+    /// # Optional Arguments
+    /// If you use [send_request](CacheClient::send_request) to set an item using a
+    /// [SetBatchRequest], you can also provide the following optional arguments:
+    ///
+    /// * `ttl` - The time-to-live for the items. If not provided, the client's default time-to-live is used.
+    ///
+    /// # Examples
+    /// Assumes that a CacheClient named `cache_client` has been created and is available.
+    /// ```
+    /// # fn main() -> anyhow::Result<()> {
+    /// # use momento_test_util::create_doctest_cache_client;
+    /// # tokio_test::block_on(async {
+    /// # let (cache_client, cache_name) = create_doctest_cache_client();
+    /// use std::collections::HashMap;
+    /// use momento::cache::SetResponse;
+    ///
+    /// let items = HashMap::from([("k1", "v1"), ("k2", "v2"), ("k3", "v3")]);
+    /// let results_map: HashMap<String, SetResponse> = cache_client.set_batch(&cache_name, items).await?.into();
+    /// # assert_eq!(results_map.clone().len(), 3);
+    ///
+    /// for (key, response) in results_map {
+    ///     println!("SetResponse for key {}: {:?}", key, response);
+    /// }
+    /// # Ok(())
+    /// # })
+    /// # }
+    /// ```
+    /// You can also use the [send_request](CacheClient::send_request) method to get an item using a [SetBatchRequest].
+    ///
+    /// For more examples of handling the response, see [SetBatchResponse].
+    pub async fn set_batch(
+        &self,
+        cache_name: impl Into<String>,
+        items: HashMap<impl IntoBytes + Copy, impl IntoBytes + Copy>,
```

Comment:
Opted to accept a `HashMap` of (key, value) pairs directly, but could switch to accepting a vector of pairs instead if that seems more user-friendly. 

From what I saw in the other SDKs, set_batch seemed split between a vector of pairs, a dictionary of pairs, or a type union of both.

Reply:
`+ Copy` is a very aggressive bound. I do not think you want it.
```rust
fn a(aa: impl Sized + Copy) {}

fn b() {
    // works be cause &'static str is Copy
    a("");
    let payload: Vec<u8> = Vec::new();
    // does not compile because Vec is not Copy
    a(payload)
}
```

Reply:
okay, good to know, thanks! I'll look into how to avoid that

Reply:
Similarly, you might find the ergonomics better if you use a looser bound than an explicit HashMap. That does not mean you should accept a vector of pairs either - instead, consider IntoIterator which will let people call you however they want:

```rust
fn c(aa: impl IntoIterator<Item = (&'static str, &'static str)>) {
    let my_proto_map: HashMap<_, _> = HashMap::from_iter(aa);
}

fn d() {
    c([("k1", "v1"), ("k2", "v2")]);

    let mut map = HashMap::new();
    map.insert("k1", "v1");
    map.insert("k2", "v2");
    c(map);

    let mut vec = vec![("k1", "v1"), ("k2", "v2")];
    c(vec);
}
```